library/core/src/array/mod.rs

```diff
@@ -364,3 +364,54 @@ macro_rules! array_impl_default {
 }
 
 array_impl_default! {32, T T T T T T T T T T T T T T T T T T T T T T T T T T T T T T T T}
+
+#[cfg(not(bootstrap))]
+#[lang = "array"]
+impl<T, const N: usize> [T; N] {
+    /// Returns an array of the same size as self, with `f` applied to each element.
```

Comment:
@GuillaumeGomez Do you think this could be improved?

Reply:
Since it's an `FnMut`, please explicitly document the order in which the closure will be called on the items, and add an example where that matters.  For example, something like
```
let x = [1, 2, 3];
let mut temp = 0;
let y = x.map(|v| { temp += 1; v + temp });
assert_eq!(y, [1, 4, 9]);
```